File: js-mocha/test/test_gilded_rose.js
var {expect} = require('chai');
var {Shop, Item} = require('../src/gilded_rose.js');
describe("Gilded Rose", function() {

  it("should foo", function() {
    const gildedRose = new Shop([ new Item("foo", 0, 0) ]);
    const items = gildedRose.updateQuality();
    expect(items[0].name).to.equal("foo");
  });

});

File: test_gilded_rose.py
import unittest

from gilded_rose import Item, GildedRose
from approvaltests.approvals import verify


class GildedRoseTest(unittest.TestCase):
    def test_foo(self):
        items = [
            Item("foo", 0, 0),
            Item("Aged Brie", 0, 0),
            Item("Backstage passes to a TAFKAL80ETC concert", 0, 0),
        ]
        gilded_rose = GildedRose(items)
        gilded_rose.update_quality()

        to_approve = []
        for item in items:
            to_approve.append(str(item))

        verify("\n".join(to_approve))


if __name__ == "__main__":
    unittest.main()
